### ISSUE TITLE
Change log for 3.6.0 does not mention backwards compatibility breaking change in AMPQSSLConnection signature

### ISSUE THREAD

**Issue by caricare2** [php-amqplib/php-amqplib#1174]
**Version(s) affected**: 3.6.0

**Description**  
Upon upgrading from 3.5.3 to 3.6.2 I discovered an error related to the parameter format of the AMPQSSLConnection connection signature. It turns out this changed in 3.6.0 but there is no specific mention of this backwards compatibility breaking change in the change log. I had to hunt through the code to figure out it was changed here:
https://github.com/php-amqplib/php-amqplib/commit/7e9e96d87793222d0f1429f59bfcc850d9c9066e#diff-7ca812c052f6560fc53e2ba72fd1c40927c86467f86c80e2016ab92c107fc919)

**Possible Solution**  
The change log should be updated to indicate this change so people upgrading can be prepared to adapt their code for it.

**Comment by ramunasd** [comment]
I see It's not a breaking change - You can still use connection class in the same way. The only difference is a new deprecation message, probably the one You are mentioning.

I'm closing this because there is no info about error itself, PHP env and usage. Please reopen with more context if still relevant.

**Comment by caricare2** [comment]
It's a breaking change. My apologies for not describing it in more detail, I thought the link would be sufficient information, but I should have been more explicit about the details of the issue.

The function signature for the AMQPSSLConnection construct was changed. There were originally 9 parameters, now there are 8. The original 8th parameter was previously $ssl_protocol, which was a string. That parameter was removed, and now the 8th parameter is ?AMQPConnectionConfig $config (which was originally the 9th parameter). As you might expect, this causes a TypeError to be thrown if you were previously passing that parameter instead of relying on the default value (I'm guessing the reason more people didn't run into this issue is that they probably were using the default values).

I was able to change my code to prevent the type error by removing that parameter, but this is very much a breaking change that should be listed in the change log. I can see why you might have missed it since the actual comment for the change doesn't mention the signature change either.
